e2e/pages/strategy.spec.ts

```diff
@@ -1,14 +1,80 @@
-import { test } from '@playwright/test';
-import { screenshot } from '../utils/operators';
+import { test, expect } from '@playwright/test';
+import { navigateTo, screenshot, waitFor } from '../utils/operators';
 import { mockApi } from '../utils/mock-api';
+import { setupImposter } from '../utils/debug';
 
 test.describe('Strategies', () => {
-  test.beforeEach(({ page }) => {
-    mockApi(page);
+  test.beforeEach(async ({ page }) => {
+    await Promise.all([mockApi(page), setupImposter(page)]);
+    await mockApi(page);
   });
   test('First Strategy Page', async ({ page }) => {
-    await page.goto('/');
+    await navigateTo(page, '/');
     await page.getByTestId('first-strategy').waitFor({ state: 'visible' });
     await screenshot(page, 'first-strategy');
   });
+
+  test('Create Limit Strategy ETH->DAI', async ({ page }) => {
+    await navigateTo(page, '/');
+    await page.getByTestId('create-strategy-desktop').click();
+
+    // Select Base
+    await page.getByTestId('select-base-token').click();
```

Comment:
Making these tests more data driven will make them reusable.
This test can be made more generic so that it can test any pair, and so you can take the following approach (pseudo code):
```js
const testCases = [
  {
    base: 'ETH',
    quote:  'DAI',
    buyLimit: 1500,
    buyBudget: 4000,
   .... //etc
  }
];

testCases.forEach(case => test(`Create Limit Strategy ${case.base}->${case.quote}`, ...
...

```

it will make it much easier to quickly add additional pairs and test cases

Reply:
I agree, but before jumping on a refactoring I would first do 2 more tests to see what is common or not.

Reply:
Alright, let's build a utils for that

Reply:
ok, decided to refactor once patterns emerge